kernel/seccomp.c

```diff
@@ -1115,6 +1115,13 @@ static int seccomp_do_user_notification(int this_syscall,
 			mutex_unlock(&match->notify_lock);
 			goto wait;
 		}
+
+		/* Allocate an fd and return it */
+		if (addfd && n.state == SECCOMP_NOTIFY_REPLIED) {
+			seccomp_handle_addfd(addfd);
+			n.val = addfd->ret;
```

Comment:
The fd allocation can fail (like max open fds is reached for the process, `seccomp_handle_addfd()` honours that) and we should probably make `v.error = 1` in that case. Will test it and add it later

Reply:
Added an if to properly handle this error